pkg/retriever/retriever.go

```diff
@@ -94,14 +96,17 @@ func NewRetriever(
 		eventManager: events.NewEventManager(ctx),
 		spTracker:    newSpTracker(nil),
 	}
-	executor := &Executor{
+	graphsyncRetriever := &GraphSyncRetriever{
 		GetStorageProviderTimeout:   retriever.getStorageProviderTimeout,
 		IsAcceptableStorageProvider: retriever.isAcceptableStorageProvider,
 		IsAcceptableQueryResponse:   retriever.isAcceptableQueryResponse,
 		Client:                      client,
 	}
+	bitswapRetriever := NewBitswapRetriever()
+	protocolSplitter := NewProtocolSplitter([]multicodec.Code{multicodec.TransportGraphsyncFilecoinv1, multicodec.TransportBitswap})
 	retrievalCandidateFinder := NewRetrievalCandidateFinder(candidateFinder, retriever.isAcceptableStorageProvider)
-	retriever.executor = types.WithCandidates(retrievalCandidateFinder.FindCandidates, executor.RetrieveFromCandidates)
+	executor := util.MultiRetriever(protocolSplitter.SplitCandidates, []types.CandidateRetriever{graphsyncRetriever.RetrieveFromCandidates, bitswapRetriever.Retrieve}, types.RaceCoordination)
+	retriever.executor = util.WithCandidates(retrievalCandidateFinder.FindCandidates, executor)
```

Comment:
The `WithCandidates` and `WithFoundCandidates` makes sense, but how they are used is confusing IMO. I don't necessarily know what's going on when I read this. The retriever executor is set to the `MultiRetriever` and then passed to `WithCandidates` and set on the retriever executor. Are there better names for these? Is there a different pattern we can use to set these?